go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -1203,6 +1220,278 @@ func (ts *trafficSwitcher) isSequenceParticipating(ctx context.Context) (bool, e
+	vschema, err := ts.TopoServer().GetVSchema(ctx, ts.targetKeyspace)
+	if err != nil {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get vschema for target keyspace %s: %v",
+			ts.targetKeyspace, err)
+	}
+	if vschema == nil || vschema.Tables == nil || len(vschema.Tables) == 0 { // Nothing to do
+		return nil, nil
+	}
+
+	targets := maps.Values(ts.Targets())
+	if len(targets) == 0 || targets[0].GetPrimary() == nil { // This should never happen
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "no primary tablet found for target keyspace %s", ts.targetKeyspace)
+	}
+	targetDBName := targets[0].GetPrimary().DbName()
+	sequencesByBackingTable := make(map[string]*sequenceMetadata)
+	smMu := sync.Mutex{}
+	for _, table := range ts.Tables() {
+		vs, ok := vschema.Tables[table]
+		if !ok || vs == nil {
+			continue
+		}
+		if vs.AutoIncrement != nil && vs.AutoIncrement.Sequence != "" {
+			sm := &sequenceMetadata{
+				backingTableName:     vs.AutoIncrement.Sequence,
+				usingTableName:       table,
+				usingTableDefinition: vs,
+				usingTableDBName:     targetDBName,
+			}
+			// If the sequence table is fully qualified in the vschema then
+			// we don't need to find it later.
+			if strings.Contains(vs.AutoIncrement.Sequence, ".") {
+				parts := strings.Split(vs.AutoIncrement.Sequence, ".")
+				if len(parts) != 2 {
+					return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "invalid sequence table name %s defined in the %s keyspace",
+						vs.AutoIncrement.Sequence, ts.targetKeyspace)
+				}
+				sm.backingTableName = parts[1]
+				sm.backingTableKeyspace = parts[0]
+			}
+			sequencesByBackingTable[sm.backingTableName] = sm
+		}
+	}
+	if len(sequencesByBackingTable) == 0 { // Nothing to do
+		return nil, nil
+	}
+
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
+
+	// If all of the sequence tables were defined using qualified table
+	// names in the vschema, then we don't need to look for them.
+	mustSearch := false
+	for _, sm := range sequencesByBackingTable {
+		if sm.backingTableKeyspace == "" {
+			mustSearch = true
+			break
+		}
+	}
+	if !mustSearch {
+		return sequencesByBackingTable, nil
+	}
+
+	// Now we need to locate the backing sequence table(s) which will
+	// be in another unsharded keyspace.
+	tableCount := len(sequencesByBackingTable)
+	tablesFound := 0                                                      // Used to short circuit the search
+	searchCompleted := make(chan struct{})                                // The search has completed
+	searchKeyspace := func(sctx context.Context, keyspace string) error { // The function used to search each keyspace
+		kvs, kerr := ts.TopoServer().GetVSchema(ctx, keyspace)
+		if kerr != nil {
+			return vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get vschema for keyspace %s: %v",
+				keyspace, kerr)
+		}
+		if kvs == nil || kvs.Sharded || kvs.Tables == nil || len(kvs.Tables) == 0 {
```

Comment:
```suggestion
		if kvs == nil || kvs.Sharded || len(kvs.Tables) == 0 {
```

Same thing as before.